### PR TITLE
[CELEBORN-595] Rename and refactor the configuration doc.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Rename and refactor the configuration doc.

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @pan3793 @FMX @RexXiong  Could you take a. look a have a disscuss?

**Review by FMX — COMMENTED** [review]
I think the consistency can be improved.

**Review by FMX — COMMENTED** [review]
I think the consistency can be improved.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -1667,7 +1887,8 @@ object CelebornConf extends Logging {
       .createWithDefaultString("4MB")
 
   val HA_MASTER_RATIS_LOG_APPENDER_QUEUE_NUM_ELEMENTS: ConfigEntry[Int] =
-    buildConf("celeborn.ha.master.ratis.raft.server.log.appender.buffer.element-limit")
+    buildConf("celeborn.ha.master.ha.ratis.raft.server.log.appender.buffer.element-limit")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1890]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -2251,183 +2505,45 @@ object CelebornConf extends Logging {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("10min")
 
-  val METRICS_APP_TOP_DISK_USAGE_COUNT: ConfigEntry[Int] =
-    buildConf("celeborn.metrics.app.topDiskUsage.count")
+  val MASTER_APP_TOP_DISK_USAGE_COUNT: ConfigEntry[Int] =
+    buildConf("celeborn.master.app.topDiskUsage.count")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2509]: Yea, I think so.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -2537,23 +2653,15 @@ object CelebornConf extends Logging {
...
       .categories("client")
       .version("0.3.0")
       .doc("Timeout for LifecycleManager request reserve slots.")
       .fallbackConf(RPC_ASK_TIMEOUT)
 
   val REGISTER_SHUFFLE_RPC_ASK_TIMEOUT: ConfigEntry[Long] =
-    buildConf("celeborn.rpc.registerShuffle.askTimeout")
+    buildConf("celeborn.client.rpc.registerShuffle.askTimeout")
       .categories("client")
       .version("0.2.0")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2666]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 4]
```diff
@@ -3124,7 +3227,8 @@ object CelebornConf extends Logging {
       .createWithDefault(10000)
 
   val COLUMNAR_SHUFFLE_OFF_HEAP_ENABLED: ConfigEntry[Boolean] =
-    buildConf("celeborn.columnar.offHeap.enabled")
+    buildConf("celeborn.columnar.shuffle.offHeap.enabled")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:3230]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 5]
```diff
@@ -1370,23 +1430,181 @@ object CelebornConf extends Logging {
...
+      .categories("worker", "test")
+      .version("0.2.0")
+      .doc("Whether to test push master data timeout")
+      .booleanConf
+      .createWithDefault(false)
+
+  val TEST_PUSH_SLAVE_DATA_TIMEOUT: ConfigEntry[Boolean] =
+    buildConf("celeborn.test.pushSlaveDataTimeout")
+      .internal
+      .categories("worker", "test")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1525]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 6]
```diff
@@ -1370,23 +1430,181 @@ object CelebornConf extends Logging {
...
+      .withAlternative("rss.push.data.rpc.timeout")
+      .categories("client")
+      .version("0.2.0")
+      .doc(s"Timeout for a task to push data rpc message. This value should better be more than twice of `${PUSH_TIMEOUT_CHECK_INTERVAL.key}`")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .checkValue(_ > 0, "celeborn.client.push.data.timeout must be positive!")
+      .createWithDefaultString("120s")
+
+  val TEST_PUSH_MASTER_DATA_TIMEOUT: ConfigEntry[Boolean] =
+    buildConf("celeborn.test.pushMasterDataTimeout")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 7]
```diff
@@ -1992,7 +2239,8 @@ object CelebornConf extends Logging {
       .createWithDefaultString("1000ms")
 
   val WORKER_HEARTBEAT_TIMEOUT: ConfigEntry[Long] =
-    buildConf("celeborn.worker.heartbeat.timeout")
+    buildConf("celeborn.master.worker.heartbeat.timeout")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Change to `celeborn.master.heartbeat.worker.timeout` & `celeborn.master.heartbeat.application.timeout`

**Comment by AngersZhuuuu** [comment]
ping  @FMX @pan3793 @RexXiong @nafiyAix @waitinfuture Could you take a look?

File: common/src/main/scala/org/apache/celeborn/common/internal/config/ConfigEntry.scala [hunk 8]
```diff
@@ -19,6 +19,7 @@ package org.apache.celeborn.common.internal.config
 
 import java.util.concurrent.ConcurrentHashMap
 
+import org.apache.celeborn.common.CelebornConf
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/internal/config/ConfigEntry.scala:22]: Yea, should remove

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 9]
```diff
@@ -1247,6 +1249,18 @@ object CelebornConf extends Logging {
       .booleanConf
       .createWithDefault(false)
 
+  val MAX_CHUNKS_BEING_TRANSFERRED: ConfigEntry[Long] =
+    buildConf("celeborn.<module>.io.maxChunksBeingTransferred")
+      .withAlternative("rss.shuffle.maxChunksBeingTransferred")
+      .categories("network")
+      .doc("The max number of chunks allowed to be transferred at the same time on shuffle service. Note " +
+        "that new incoming connections will be closed when the max number is hit. The client will retry " +
+        "according to the shuffle retry configs (see `celeborn.shuffle.io.maxRetries` and " +
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1258]: > `celeborn.shuffle.io.maxRetries` does not exist.

Now we use `celeborn.<module>.io.maxRetries`, changedd

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 10]
```diff
@@ -1247,6 +1249,18 @@ object CelebornConf extends Logging {
       .booleanConf
       .createWithDefault(false)
 
+  val MAX_CHUNKS_BEING_TRANSFERRED: ConfigEntry[Long] =
+    buildConf("celeborn.<module>.io.maxChunksBeingTransferred")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1253]: > Is there more than one `<module>` here?

I know it only used in fetch module, but we can name it follow the transport conf's configuration format.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 11]
```diff
@@ -1469,7 +1733,8 @@ object CelebornConf extends Logging {
       .createWithDefault(false)
 
   val WORKER_EXCLUDED_EXPIRE_TIMEOUT: ConfigEntry[Long] =
-    buildConf("celeborn.worker.excluded.expireTimeout")
+    buildConf("celeborn.client.worker.excluded.expireTimeout")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1736]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 12]
```diff
@@ -1485,18 +1750,31 @@ object CelebornConf extends Logging {
...
+      .categories("client")
+      .version("0.2.0")
+      .doc("Max chunk size of reducer's merged shuffle data. For example, if a reducer's " +
+        "shuffle data is 128M and the data will need 16 fetch chunk requests to fetch.")
+      .bytesConf(ByteUnit.BYTE)
+      .createWithDefaultString("8m")
+
   val SHUFFLE_CHUNK_SIZE: ConfigEntry[Long] =
-    buildConf("celeborn.shuffle.chunk.size")
+    buildConf("celeborn.worker.shuffle.chunk.size")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1765]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 13]
```diff
@@ -1485,18 +1750,31 @@ object CelebornConf extends Logging {
       .booleanConf
       .createWithDefault(true)
 
+  val DFS_SHUFFLE_CHUNK_SIZE: ConfigEntry[Long] =
+    buildConf("celeborn.client.shuffle.dfs.read.chunk.size")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1754]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 14]
```diff
@@ -1309,16 +1314,74 @@ object CelebornConf extends Logging {
...
+      .withAlternative("rss.reserve.slots.retry.wait")
+      .categories("client")
+      .version("0.2.0")
+      .doc("Wait time before next retry if reserve slots failed.")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("3s")
+
+  val COMMIT_FILE_REQUEST_MAX_RETRY: ConfigEntry[Int] =
+    buildConf("celeborn.client.rpc.requestCommitFiles.maxRetries")
+      .withAlternative("celeborn.rpc.requestCommitFiles.maxRetries")
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 15]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
 
   val HA_MASTER_RATIS_STORAGE_DIR: ConfigEntry[String] =
     buildConf("celeborn.ha.master.ratis.raft.server.storage.dir")
-      .categories("master")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.storage.dir")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1872]: identical

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 16]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
+      .categories("ha")
       .withAlternative("rss.ha.storage.dir")
       .version("0.2.0")
       .stringConf
       .createWithDefault("/tmp/ratis")
 
   val HA_MASTER_RATIS_LOG_SEGMENT_SIZE_MAX: ConfigEntry[Long] =
     buildConf("celeborn.ha.master.ratis.raft.server.log.segment.size.max")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.log.segment.size.max")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1881]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 17]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
       .internal
-      .categories("master")
+      .categories("ha")
       .version("0.2.0")
       .bytesConf(ByteUnit.BYTE)
       .createWithDefaultString("4MB")
 
   val HA_MASTER_RATIS_LOG_PREALLOCATED_SIZE: ConfigEntry[Long] =
     buildConf("celeborn.ha.master.ratis.raft.server.log.preallocated.size")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.log.preallocated.size")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1891]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 18]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
       .internal
-      .categories("master")
+      .categories("ha")
       .version("0.2.0")
       .bytesConf(ByteUnit.BYTE)
       .createWithDefaultString("4MB")
 
   val HA_MASTER_RATIS_LOG_APPENDER_QUEUE_NUM_ELEMENTS: ConfigEntry[Int] =
     buildConf("celeborn.ha.master.ratis.raft.server.log.appender.buffer.element-limit")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.log.appender.buffer.element-limit")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1901]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 19]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
       .internal
-      .categories("master")
+      .categories("ha")
       .version("0.2.0")
       .intConf
       .createWithDefault(1024)
 
   val HA_MASTER_RATIS_LOG_APPENDER_QUEUE_BYTE_LIMIT: ConfigEntry[Long] =
     buildConf("celeborn.ha.master.ratis.raft.server.log.appender.buffer.byte-limit")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.log.appender.buffer.byte-limit")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1911]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 20]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
       .internal
-      .categories("master")
+      .categories("ha")
       .version("0.2.0")
       .booleanConf
       .createWithDefault(true)
 
   val HA_MASTER_RATIS_LOG_PURGE_GAP: ConfigEntry[Int] =
     buildConf("celeborn.ha.master.ratis.raft.server.log.purge.gap")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.log.purge.gap")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1930]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 21]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
       .internal
-      .categories("master")
+      .categories("ha")
       .version("0.2.0")
       .intConf
       .createWithDefault(1000000)
 
   val HA_MASTER_RATIS_RPC_REQUEST_TIMEOUT: ConfigEntry[Long] =
     buildConf("celeborn.ha.master.ratis.raft.server.rpc.request.timeout")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.rpc.request.timeout")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1940]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 22]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
       .internal
-      .categories("master")
+      .categories("ha")
       .version("0.2.0")
       .timeConf(TimeUnit.SECONDS)
       .createWithDefaultString("3s")
 
   val HA_MASTER_RATIS_SERVER_RETRY_CACHE_EXPIRY_TIME: ConfigEntry[Long] =
     buildConf("celeborn.ha.master.ratis.raft.server.retrycache.expirytime")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.retrycache.expirytime")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1950]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 23]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
       .internal
-      .categories("master")
+      .categories("ha")
       .version("0.2.0")
       .timeConf(TimeUnit.SECONDS)
       .createWithDefaultString("600s")
 
   val HA_MASTER_RATIS_RPC_TIMEOUT_MIN: ConfigEntry[Long] =
     buildConf("celeborn.ha.master.ratis.raft.server.rpc.timeout.min")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.rpc.timeout.min")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1960]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 24]
```diff
@@ -1642,120 +1869,131 @@ object CelebornConf extends Logging {
...
       .internal
-      .categories("master")
+      .categories("ha")
       .version("0.2.0")
       .timeConf(TimeUnit.SECONDS)
       .createWithDefaultString("3s")
 
   val HA_MASTER_RATIS_RPC_TIMEOUT_MAX: ConfigEntry[Long] =
     buildConf("celeborn.ha.master.ratis.raft.server.rpc.timeout.max")
+      .withAlternative("celeborn.ha.master.ratis.raft.server.rpc.timeout.max")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1970]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 25]
```diff
@@ -1964,37 +2214,11 @@ object CelebornConf extends Logging {
...
-      .withAlternative("rss.worker.checkFileCleanTimeoutMs")
-      .categories("worker")
-      .doc("The wait time per retry for a worker to check if the working directory is cleaned up before registering with the master.")
-      .version("0.2.0")
-      .timeConf(TimeUnit.MILLISECONDS)
-      .createWithDefaultString("1000ms")
-
   val WORKER_HEARTBEAT_TIMEOUT: ConfigEntry[Long] =
-    buildConf("celeborn.worker.heartbeat.timeout")
+    buildConf("celeborn.master.heartbeat.worker.timeout")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2218]: Maybe ```celeborn.heartbeat.worker.timeout```

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 26]
```diff
@@ -2010,7 +2234,8 @@ object CelebornConf extends Logging {
       .createWithDefault(64)
 
   val WORKER_COMMIT_THREADS: ConfigEntry[Int] =
-    buildConf("celeborn.worker.commit.threads")
+    buildConf("celeborn.worker.shuffle.commit.threads")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2237]: I think the origin is better or ```celeborn.worker.commitFiles.threads```. Maybe we should make a principle that there shouldn't be ```shuffle``` inside a config name.

**Comment by waitinfuture** [comment]
And we should also re-organize the configs according to categories, better to do it in follow-up PR to make it clear.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 15 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Removed, miss to delete this line

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 26 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2226]: yea

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 25 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2207]: Yea

**Comment by AngersZhuuuu** [comment]
@FMX @waitinfuture @pan3793 @RexXiong Could you take a look again?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 27]
```diff
@@ -1368,40 +1385,46 @@ object CelebornConf extends Logging {
...
       .createWithDefault(false)
 
   val PUSH_TIMEOUT_CHECK_INTERVAL: ConfigEntry[Long] =
-    buildConf("celeborn.push.timeoutCheck.interval")
-      .categories("client", "worker")
-      .doc("Interval for checking push data timeout.")
+    buildConf("celeborn.<module>.push.timeoutCheck.interval")
+      .categories("network")
+      .doc("Interval for checking push data timeout. " +
+        s"If setting `module` to `${TransportModuleConstants.DATA_MODULE}`, it works for shuffle client push data to worker." +
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1409]: and should be configured in client side.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 28]
```diff
@@ -1368,40 +1385,46 @@ object CelebornConf extends Logging {
...
+        s"If setting `module` to `${TransportModuleConstants.REPLICATE_MODULE}`, it works for worker replicate data to peer worker.")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1410]: and should be configured in worker side.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 29]
```diff
@@ -1368,40 +1385,46 @@ object CelebornConf extends Logging {
...
       .createWithDefaultString("5s")
 
   val PUSH_TIMEOUT_CHECK_THREADS: ConfigEntry[Int] =
-    buildConf("celeborn.push.timeoutCheck.threads")
-      .categories("client", "worker")
-      .doc("Threads num for checking push data timeout.")
+    buildConf("celeborn.<module>.push.timeoutCheck.threads")
+      .categories("network")
+      .doc("Threads num for checking push data timeout. " +
+        s"If setting `module` to `${TransportModuleConstants.DATA_MODULE}`, it works for shuffle client push data to worker." +
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1419]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 30]
```diff
@@ -1368,40 +1385,46 @@ object CelebornConf extends Logging {
...
+        s"If setting `module` to `${TransportModuleConstants.REPLICATE_MODULE}`, it works for worker replicate data to peer worker.")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1420]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 31]
```diff
@@ -2229,36 +2286,40 @@ object CelebornConf extends Logging {
...
       .createWithDefaultString("64mb")
 
-  val SHUFFLE_MIN_PARTITION_SIZE_TO_ESTIMATE: ConfigEntry[Long] =
-    buildConf("celeborn.shuffle.minPartitionSizeToEstimate")
+  val MIN_PARTITION_SIZE_TO_ESTIMATE: ConfigEntry[Long] =
+    buildConf("celeborn.master.estimatedPartitionSize.minSize")
+      .withAlternative("celeborn.shuffle.minPartitionSizeToEstimate")
       .withAlternative("rss.minimum.estimate.partition.size")
-      .categories("worker")
+      .categories("master")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2303]: should be worker

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 32]
```diff
@@ -3316,11 +3423,12 @@ object CelebornConf extends Logging {
       .booleanConf
       .createWithDefault(true)
 
-  val CLIENT_HEARTBEAT_INTERVAL: ConfigEntry[Long] =
-    buildConf("celeborn.client.heartbeat.interval")
-      .categories("client", "worker")
+  val CHANNEL_HEARTBEAT_INTERVAL: ConfigEntry[Long] =
+    buildConf("celeborn.<module>.heartbeat.interval")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:3427]: Will it work when configure with ```celeborn.client.heartbeat.interval``` ?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 31 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2303]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 27 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 28 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Done

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 29 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 33]
```diff
@@ -1406,8 +1406,10 @@ object CelebornConf extends Logging {
     buildConf("celeborn.<module>.push.timeoutCheck.interval")
       .categories("network")
       .doc("Interval for checking push data timeout. " +
-        s"If setting `module` to `${TransportModuleConstants.DATA_MODULE}`, it works for shuffle client push data to worker." +
-        s"If setting `module` to `${TransportModuleConstants.REPLICATE_MODULE}`, it works for worker replicate data to peer worker.")
+        s"If setting <module> to `${TransportModuleConstants.DATA_MODULE}`, " +
+        s"it works for shuffle client push and fetch data and should be configured in client side. " +
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1410]: just for push?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 34]
```diff
@@ -1416,8 +1418,10 @@ object CelebornConf extends Logging {
     buildConf("celeborn.<module>.push.timeoutCheck.threads")
       .categories("network")
       .doc("Threads num for checking push data timeout. " +
-        s"If setting `module` to `${TransportModuleConstants.DATA_MODULE}`, it works for shuffle client push data to worker." +
-        s"If setting `module` to `${TransportModuleConstants.REPLICATE_MODULE}`, it works for worker replicate data to peer worker.")
+        s"If setting <module> to `${TransportModuleConstants.DATA_MODULE}`, " +
+        s"it works for shuffle client push and fetch data and should be configured in client side. " +
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1422]: ditto

**Review by waitinfuture — APPROVED** [review]
LGTM except some nits, thanks!

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 32 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:3431]: > Will it work when configure with `celeborn.client.heartbeat.interval` ? And better to doc about the modules.

Updated

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 34 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: DOne

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 33 above)
Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: DOne

**Comment by waitinfuture** [comment]
LGTM, thanks for the PR! Merging to main.

**Comment by pan3793** [comment]
the renamed version configurations should update version as well, as we discussed during the last meeting

**Pull request merged**